doc/devel/development_setup.rst

```diff
@@ -82,6 +80,10 @@ so that you can access the latest changes there (see also `Managing remote repos
 
          git remote add upstream git@github.com:matplotlib/matplotlib.git
 
+You can now use ``upstream`` to retrieve the most current snapshot of the source
+code, as described in :ref:`development-workflow`.  See also `Managing remote repositories
+<https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories>`__.
```

Comment:
I propose to move this link to the notes section below (or remove it completely). While it is about remotes, it only covers rarely needed management operations on remotes, not something one would need for regular development work:

[Adding a remote repository](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#adding-a-remote-repository). - we've covered that ourselves
[Changing a remote repository's URL](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#changing-a-remote-repositorys-url) - the matplotlib repository won't move
[Renaming a remote repository](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories#renaming-a-remote-repository) - no need to change the name `upstream`
[Removing a remote repository](https://docs.Github.com/en/get-started/getting-started-with-git/managing-remote-repositories#removing-a-remote-repository) - no need to decouple from upstream